docs/subnets/deploy-a-gnosis-safe-on-your-evm.md

```diff
@@ -0,0 +1,810 @@
+
+Now we will use hardhat to ensure that our proxy was successfully upgraded.
+
+First, navigate back to hardhat console in your `avalanche-smart-contract-quickstart-project` and instantiate `StorageV2` at our proxy address.
+
+```zsh
+> const storageV2 = await ethers.getContractAt('StorageV2','YOUR-PROXY-ADDRESS-HERE')
+```
+
+Notice that we are now using `StorageV2` at our original proxy address. Since, we've upgraded our implementation, our we can call the original address but interact with the new contract. 
+
+Lets check the stored value to ensure that we have retained the data from the previous implementation..
+
+```zsh
+> (await storageV2.retrieve()).toString()
+'42'
+```
+
+Great! We've successfully retrieved the previously stored value from the contract!
+
+Now let's call our upgraded contract's new function [`increment`](https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/964129dfb7cb9271b396927e9ea8b009e321cda1/contracts/StorageV2.sol#L24-L27) which adds 1 to the stored value:
+
+```zsh
+> await storageV2.increment()
+{
+  hash: '0x2ed9dff3f909a50f191d41ff59ab423907fbc23b4cf7b3721907d933a710b848',
+  type: 0,
+  accessList: null,
+  blockHash: '0x70be069ddf5c353cef0f6e3047b20e9e8c52b837228c131d3e0ad8c84b4c39f4',
+  blockNumber: 124,
+  transactionIndex: 0,
+  confirmations: 1,
+  from: '0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC',
+  gasPrice: BigNumber { _hex: '0x05d21dba00', _isBigNumber: true },
+  gasLimit: BigNumber { _hex: '0x8931', _isBigNumber: true },
+  to: '0x5dda6Fa725248D95d2086F4fcEb6bA6bdfEbc45b',
+  value: BigNumber { _hex: '0x00', _isBigNumber: true },
+  nonce: 125,
+  data: '0xd09de08a',
+  r: '0xd5ee62766bf7f88946c0d565dbb90b80e2a93df42137b1c13ac44808f7727297',
+  s: '0x4f46a263fdd6f5518ca2cfd43440ee14b5dfec2322e045a3eebdd1d51558c6db',
+  v: 200034,
+  creates: null,
+  chainId: 99999,
+  wait: [Function (anonymous)]
+}
+```
+
+The transaction data shows us that the operation was successful!
+
+Now, lets check the stored number.
+
+```zsh
+> (await storageV2.retrieve()).toString()
+'43'
+```
+
+And there you have it. 
+We have successfully done the following:
+- Deployed a [Transparent Upgradeable Proxy](https://blog.openzeppelin.com/the-transparent-proxy-pattern/).
+- [Transferred proxy admin ownership](https://docs.openzeppelin.com/contracts/4.x/api/access#Ownable-transferOwnership-address-) to a Gnosis Safe.
+- [Upgraded our proxy](https://docs.openzeppelin.com/contracts/4.x/api/proxy#ProxyAdmin-upgrade-contract-TransparentUpgradeableProxy-address-) to a new implementation.
+
+## Local Workflow
+
+### Start the Local Network
+
+Follow [Create a Local Test Network](../quickstart/create-a-local-test-network.md#avalanche-network-runner) to start a 5-node local network. Make sure that you get one of the port numbers by following [this](../quickstart/create-a-local-test-network.md#retrieve-all-nodes). In this tutorial, we will assume one of the ports is 49435.
```

Comment:
Should https://docs.avax.network/subnets/create-a-local-subnet be referenced to creat a subnet, rather on the primary network?

Reply:
Yeah it should. It makes sense to highlight it